src/test/java/ch/uzh/ifi/hase/soprafs23/service/LobbyServiceTest.java

```diff
@@ -0,0 +1,197 @@
+import ch.uzh.ifi.hase.soprafs23.entity.Lobby;
+import ch.uzh.ifi.hase.soprafs23.entity.User;
+import ch.uzh.ifi.hase.soprafs23.repository.LobbyRepository;
+import ch.uzh.ifi.hase.soprafs23.repository.UserRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class LobbyServiceTest {
+
+    @Mock
+    private LobbyRepository lobbyRepository;
+
+    @InjectMocks
+    private LobbyService lobbyService;
+    @Mock
+    private UserRepository userRepository;
+
+    @InjectMocks
+    private UserService userService;
+    private Lobby testLobby;
+    private User testUser;
+
+    @BeforeEach
+    public void setup() {
+        MockitoAnnotations.openMocks(this);
+        testLobby = new Lobby();
+        testLobby.setAccessCode(123456);
+        testLobby.setSettings(new Settings());
+
+        testUser = new User();
+        testUser.setId(1L);
+        testUser.setLeader(true);
+        testUser.setUsername("testUsername");
+
+        // when -> any object is being saved in the userRepository -> return the dummy
+        // testUser
+        Mockito.when(userRepository.save(Mockito.any())).thenReturn(testUser);
+
+        Mockito.when(lobbyRepository.save(Mockito.any())).thenReturn(testLobby);
+    }
+
+    @Test
+    public void createAccessCode_noDuplicate() {
+        int testCode1 = lobbyService.createAccessCode();
+        int testCode2 = lobbyService.createAccessCode();
+        int testCode3 = lobbyService.createAccessCode();
+
+        assertNotEquals(testCode1,testCode2);
+        assertNotEquals(testCode3,testCode2);
+    }
+    @Test
+    void createLobby_validInput(){
+        Lobby lobby = lobbyService.createLobby();
+        Mockito.verify(lobbyRepository, Mockito.times(1)).save(Mockito.any());
+
+        assertEquals(lobby.getSettings(),testLobby.getSettings());
+        assertEquals(lobby.getAccessCode(),testLobby.getAccessCode());
+
+    }
+    @Test
+    public void joinLobbyTeam_LobbyDoesntExist() {
+        assertThrows(ResponseStatusException.class, () -> lobbyService.joinLobbyTeam(123445, 1, 1));
+    }
+    @Test
+    public void joinLobbyTeam_UserDoesntExist() {
+        Lobby lobby = new Lobby();
+        int code = lobby.getAccessCode();
+        assertThrows(ResponseStatusException.class, () -> lobbyService.joinLobbyTeam(code, 1, 1));
+    }
+    @Test
+    public void joinLobbyTeam_UserNotInLobby() {
+        Lobby lobby = new Lobby();
+        User user =new User();
+        Long id = user.getId();
+        int code = lobby.getAccessCode();
+        assertThrows(NullPointerException.class, () -> lobbyService.joinLobbyTeam(code, 1, Math.toIntExact(id)));
```

Comment:
Does standard type casting not work? Something like:
```
(int) id
```

Reply:
no